src/tests/Interop/COM/Activator/Program.cs

```diff
@@ -32,29 +32,35 @@ static void InvalidInterfaceRequest()
                 "Non-IClassFactory request should fail");
         }
 
-        static void NonrootedAssemblyPath()
+        static void NonrootedAssemblyPath(bool builtInComDisabled)
         {
             Console.WriteLine($"Running {nameof(NonrootedAssemblyPath)}...");
 
-            ArgumentException e = Assert.Throws<ArgumentException>(
-                () =>
+            Action action = () =>
                 {
                     var cxt = new ComActivationContext()
                     {
                         InterfaceId = typeof(IClassFactory).GUID,
                         AssemblyPath = "foo.dll"
                     };
                     ComActivator.GetClassFactoryForType(cxt);
-                },
-                "Non-root assembly path should not be valid");
+                };
+
+            if (!builtInComDisabled)
+            {
+                Assert.Throws<ArgumentException>(action, "Non-root assembly path should not be valid", AssertThrowsOptions.AllowDerived);
+            }
+            else
+            {
+                Assert.Throws<NotSupportedException>(action, "Built-in COM has been disabled via a feature switch", AssertThrowsOptions.AllowDerived);
```

Comment:
We shouldn't have the `AssertThrowsOptions.AllowDerived`, since you went with adding the action variable and keeping the assert for the derived exception. Sorry for not being clear before - using that option was only needed if you switched to `Assert.Throws<Exception>` and store the exception to validate it.